flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/com/ververica/cdc/connectors/doris/sink/DorisEventSerializer.java

```diff
@@ -0,0 +1,134 @@
+                }
+                schemaMaps.put(
+                        tableId,
+                        SchemaUtils.applySchemaChangeEvent(
+                                schemaMaps.get(tableId), schemaChangeEvent));
+            }
+        }
+        return null;
+    }
+
+    private Tuple2<String, byte[]> applyDataChangeEvent(DataChangeEvent event)
+            throws JsonProcessingException {
+        TableId tableId = event.tableId();
+        Schema schema = schemaMaps.get(tableId);
+        Preconditions.checkNotNull(schema, event.tableId() + " is not existed");
+        String tableKey = String.format("%s.%s", tableId.getSchemaName(), tableId.getTableName());
+        Map<String, Object> valueMap;
+        OperationType op = event.op();
+        switch (op) {
+            case INSERT:
+            case UPDATE:
+            case REPLACE:
+                valueMap = serializerRecord(event.after(), schema);
+                addDeleteSign(valueMap, false);
+                break;
+            case DELETE:
+                valueMap = serializerRecord(event.before(), schema);
+                addDeleteSign(valueMap, true);
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupport Operation " + op);
+        }
+        return Tuple2.of(
+                tableKey,
+                objectMapper.writeValueAsString(valueMap).getBytes(StandardCharsets.UTF_8));
+    }
+
+    /** serializer RecordData to Doris Value. */
+    public Map<String, Object> serializerRecord(RecordData recordData, Schema schema) {
+        List<Column> columns = schema.getColumns();
+        Map<String, Object> record = new HashMap<>();
+        Preconditions.checkState(
+                columns.size() == recordData.getArity(),
+                "Column size does not match the data size");
+
+        List<RecordData.FieldGetter> fieldGetters = SchemaUtils.createFieldGetters(schema);
+        GenericRowData rowData = toFlinkRowData(recordData, fieldGetters);
+        for (int i = 0; i < recordData.getArity(); i++) {
+            DataType dataType = DataTypeUtils.toFlinkDataType(columns.get(i).getType());
+            DorisRowConverter.SerializationConverter converter =
+                    DorisRowConverter.createNullableExternalConverter(dataType.getLogicalType());
+            Object field = converter.serialize(i, rowData);
+            record.put(columns.get(i).getName(), field);
+        }
+        return record;
+    }
+
+    /** convert recordData to Flink GenericRowData. */
+    public static GenericRowData toFlinkRowData(
+            RecordData recordData, List<RecordData.FieldGetter> fieldGetters) {
+        Preconditions.checkState(fieldGetters.size() == recordData.getArity());
+        GenericRowData rowData = new GenericRowData(recordData.getArity());
+        for (int i = 0; i < recordData.getArity(); i++) {
+            Object field = fieldGetters.get(i).getFieldOrNull(recordData);
```

Comment:
the Object from FieldGetters.getFieldOrNull will be under `com.ververica.cdc.common.data` package, This may not be as expected？

Reply:
Thanks for reminding me, I have modified it